core/chains/sui.go

```diff
@@ -0,0 +1,117 @@
+package chains
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"math/big"
+	"net/http"
+	"sort"
+	"strconv"
+	"time"
+
+	utils "github.com/xenowits/nakamoto-coefficient-calculator/core/utils"
+)
+
+type SuiResponse struct {
+	Result struct {
+		ActiveValidators []struct {
+			VotingPower string `json:"votingPower"`
+		} `json:"activeValidators"`
+	} `json:"result"`
+}
+
+type rawBody struct {
+	JSONRPC string        `json:"jsonrpc"`
+	ID      int           `json:"id"`
+	Method  string        `json:"method"`
+	Params  []interface{} `json:"params"`
+}
+
+func Sui() (int, error) {
+	request := rawBody{
+		JSONRPC: "2.0",
+		ID:      1,
+		Method:  "suix_getLatestSuiSystemState",
+		Params:  []interface{}{},
+	}
+
+	url := "https://fullnode.mainnet.sui.io"
+
+	return FetchDataSUI("sui", url, request)
+}
+
+func FetchDataSUI(chainName string, url string, request rawBody) (int, error) {
```

Comment:
can keep this function private
```suggestion
func fetchDataSUI(chainName string, url string, request rawBody) (int, error) {
```